### PR TITLE
chore: upgrade react version

### PR DESCRIPTION
**Pull request by rogeriochaves**
This merges the PRs #87 and #66

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -3,7 +3,8 @@
   <head>
     <link rel="stylesheet" type="text/css" href="https://eafdbc63c97ce6bec9ef-b0a668e5876bef6fe25684caf71db405.ssl.cf1.rackcdn.com/v1-latest/canon.min.css">
     <script type="application/javascript" src="https://eafdbc63c97ce6bec9ef-b0a668e5876bef6fe25684caf71db405.ssl.cf1.rackcdn.com/v1-latest/canon.min.js"></script>
-    <script src="https://fb.me/react-with-addons-0.13.3.min.js"></script>
+    <script src="https://fb.me/react-0.14.8.min.js"></script>
```
Comment (wpwood) [inline demo/index.html]: Not React 15? Is this because canon-react has to support previous React versions?

File: test/ButtonSpec.jsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import Button from '../transpiled/Button';
-import React from 'react/addons';
+import React from 'react';
+import ReactDOM from 'react-dom';
 let TestUtils = React.addons.TestUtils;
```
Comment (wpwood) [inline test/ButtonSpec.jsx]: Should this be 

```
import TestUtils from 'react-addons-test-utils';
```

like above?

File: demo/index.html (hunk 1 above)
Comment (adampickeral) [inline demo/index.html]: discussed IRL: we'll make one release that supports 0.14.8 then another one (with major version change) that supports 15

File: test/ButtonSpec.jsx (hunk 2 above)
Comment (rogeriochaves) [inline test/ButtonSpec.jsx]: yes, this spec is the only one that does not work when I use the new test utils, still trying to figure out why while you review the rest of the pr :)

Comment (wpwood) [inline test/ButtonSpec.jsx]: Gotcha :)

**Comment by wpwood** [comment]
Looks reasonable to me

:facepunch: :boom:

**Comment by eduaquiles** [comment]
:+1:

**Pull request merged**